pkg/instancetype/instancetype.go

```diff
@@ -954,29 +987,44 @@ func applyCPU(field *k8sfield.Path, instancetypeSpec *instancetypev1beta1.Virtua
 		vmiSpec.Domain.CPU.Realtime = instancetypeSpec.CPU.Realtime.DeepCopy()
 	}
 
+	applyGuestCPUTopology(instancetypeSpec.CPU.Guest, preferenceSpec, vmiSpec)
+
+	return nil
+}
+
+func applyGuestCPUTopology(vCPUs uint32, preferenceSpec *instancetypev1beta1.VirtualMachinePreferenceSpec, vmiSpec *virtv1.VirtualMachineInstanceSpec) {
 	// Apply the default topology here to avoid duplication below
 	vmiSpec.Domain.CPU.Cores = 1
 	vmiSpec.Domain.CPU.Sockets = 1
 	vmiSpec.Domain.CPU.Threads = 1
 
+	if vCPUs == 1 {
+		return
+	}
+
 	switch GetPreferredTopology(preferenceSpec) {
 	case instancetypev1beta1.PreferCores:
-		vmiSpec.Domain.CPU.Cores = instancetypeSpec.CPU.Guest
+		vmiSpec.Domain.CPU.Cores = vCPUs
 	case instancetypev1beta1.PreferSockets, instancetypev1beta1.PreferAny:
-		vmiSpec.Domain.CPU.Sockets = instancetypeSpec.CPU.Guest
+		vmiSpec.Domain.CPU.Sockets = vCPUs
 	case instancetypev1beta1.PreferThreads:
-		vmiSpec.Domain.CPU.Threads = instancetypeSpec.CPU.Guest
+		vmiSpec.Domain.CPU.Threads = vCPUs
 	case instancetypev1beta1.PreferSpread:
-		spreadRatio := preferenceSpec.PreferSpreadSocketToCoreRatio
-		if spreadRatio == 0 {
-			spreadRatio = DefaultSpreadRatio
+		ratio, across := GetSpreadOptions(preferenceSpec)
+		switch across {
+		case instancetypev1beta1.SpreadAcrossSocketsCores:
+			vmiSpec.Domain.CPU.Cores = ratio
+			vmiSpec.Domain.CPU.Sockets = vCPUs / ratio
```

Comment:
`uint32` so it rounds to full number of Sockets resp. does not return float on division?

Reply:
ACK